### PR TITLE
Add notify-errbot rule to ChatOps pack.

### PR DESCRIPTION
**Pull request by nzlosh**
Add notify-errbot rule to chatops pack to simplify installation for err-stackstorm users. (transpartent for non err-stackstorm users).

Updated readme using pack2md as well as fix file permissions and the metadata for route field definition.

### CODE REVIEW

File: contrib/chatops/pack.yaml
Comment (nmaludy) [inline contrib/chatops/pack.yaml:4]: Probably bump the version to 1.1.0 because it's a feature add.

File: contrib/chatops/README.jinja
Comment (nmaludy) [inline contrib/chatops/README.jinja:1]: curious how you used this to generate the readme?

File: contrib/chatops/CHANGES.md
Comment (nmaludy) [inline contrib/chatops/CHANGES.md:2]: Can you please also add a CHANGELOG entry into the root CHANGELOG.rst so it's known to a wider audience and can be captured in the release notes for StackStorm?

File: contrib/chatops/pack.yaml
Comment (nzlosh) [reply contrib/chatops/pack.yaml:4]: Sounds reasonable to me.  I'll bump the minor version.

File: contrib/chatops/README.jinja
Comment (nzlosh) [reply contrib/chatops/README.jinja:1]: I used pack2md https://github.com/nzlosh/pack2md.  There's a footnote in the README.md

File: contrib/chatops/CHANGES.md
Comment (nzlosh) [reply contrib/chatops/CHANGES.md:2]: Will do

**Comment by nzlosh** [comment]
Anything else to get this merged to master?

File: contrib/chatops/CHANGES.md
Comment (nmaludy) [reply contrib/chatops/CHANGES.md:2]: @nzlosh still need changelog in root CHANGELOG.rst

**Review by nmaludy — APPROVED** [review]

**Comment by nmaludy** [comment]
@nzlosh can you force push to this branch to kick off e2e tests again please?

**Review by armab — CHANGES_REQUESTED** [review]
I left a few comments to address before we can merge the PR.

I have no problem with adding the errbot rule, however the PR does too much and goes further in a way that might be controversial. I also felt like this PR highlights the names and the URLs more, rather then adding code.

Can we focus on the actual `rule` instead?

File: CHANGELOG.rst
Comment (nzlosh) [reply CHANGELOG.rst:42]: This pull request was initially for v3.3.0, I'll update the comment to be part of `in development`

File: contrib/chatops/README.md
Comment (nzlosh) [reply contrib/chatops/README.md:1]: I'll remove the readme update.

File: contrib/chatops/README.md
Comment (nzlosh) [reply contrib/chatops/README.md:6]: Making a change to the chatops pack, it seems like a bold statement to claim to be a coder owner of st2.  Perhaps each pack should have it's own `OWNERS.md` file to clarify this?

File: contrib/chatops/README.md
Comment (nzlosh) [reply contrib/chatops/README.md:3]: It is helpful, but not part of the `pack.yaml`.  pack2md removed it as part of the template.  This change will be reverted.

File: contrib/chatops/README.md
Comment (nzlosh) [reply contrib/chatops/README.md:9]: You are seeing the inconsistency between action descriptions and documentation.  This is one of the key advantages of pack2md for pack developers: it promotes the DRY principal.  You define your documentation in the action metadata and you get user reference documentation from pack2md.  This change is being reverted.

File: contrib/chatops/rules/notify_errbot.yaml
Comment (nzlosh) [reply contrib/chatops/rules/notify_errbot.yaml:5]: errbot will be included in the description.

**Review by armab — APPROVED** [review]
Thanks a lot for the changes :+1: 

We'll merge it today.

File: contrib/chatops/README.md
Comment (armab) [reply contrib/chatops/README.md]: https://github.com/StackStorm/st2/blob/master/OWNERS.md highlights the areas of expertise and responsibilities for the Maintainers and Contributors around StackStorm codebase in general, if it's Chatops, CI, Orquestra or any other systems.

https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners#example-of-a-codeowners-file is used as well for the projects.

**Comment by armab** [comment]
Looks like the e2e tests are failing for real in this PR.

st2 `e2e` tests failure:
```
ok 1 StackStorm's client connection
ok 2 npm directory exists
ok 3 some StackStorm aliases are enabled
not ok 4 chatops.notify rule exists
# (from function `assert_output' in file chatops/../test_helpers/bats-assert/src/assert.bash, line 239,
#  in test file chatops/test_hubot.bats, line 36)
#   `assert_output \"chatops.notify\"' failed
# 
# -- output differs --
# expected (1 lines):
#   chatops.notify
# actual (2 lines):
#   chatops.notify
#   chatops.notify-errbot
# --
# 
ok 5 hubot help command works
ok 6 chatops.post_message execution and receive status works
ok 7 complete request-response flow"
```

Here is the actual test case: https://github.com/StackStorm/st2tests/blob/6f384e7a03ac55d6b57d3036812b2fd01660ac3e/chatops/test_hubot.bats#L31-L42

@nzlosh Looks like more work needs to be done in `st2tests` repository as well. Can you provide a PR there?

**Comment by nzlosh** [comment]
I've updated the st2test repo to test for the presence of the notify-errbot rule in the chatops pack.

**Comment by blag** [comment]
The `contrib/chatops/rules/notify_hubot.yaml` file was chmodded: `100755 → 100644` - was this intentional or can we undo that?

**Comment by blag** [comment]
Tests should be partially fixed when we merge in StackStorm/st2tests#194.

Once that is merged and these tests are passing, we can merge this in.

Once this is merged in, we can merge in StackStorm/st2tests#195 to check for the additional chatops notification channel rule.

**Comment by armab** [comment]
Hm, after re-running e2e tests after https://github.com/StackStorm/st2tests/pull/194 they still fail with the same error:

```
ok 1 StackStorm's client connection
ok 2 npm directory exists
ok 3 some StackStorm aliases are enabled
not ok 4 chatops.notify rule exists
# (from function `assert_output' in file chatops/../test_helpers/bats-assert/src/assert.bash, line 239,
#  in test file chatops/test_hubot.bats, line 41)
#   `assert_output \"true\"' failed
# 
# -- output differs --
# expected (1 lines):
#   true
# actual (2 lines):
#   true
#   true
# --
# 
ok 5 hubot help command works
ok 6 chatops.post_message execution and receive status works
ok 7 complete request-response flow"
```

From what I understand the tests for this PR are cloned from the master branch of https://github.com/StackStorm/st2tests/
@blag any ideas?

**Comment by nzlosh** [comment]
@blag `contrib/chatops/rules/notify_hubot.yaml` file was chmodded: `100755 → 100644`, yes that was intentional on my part, yaml is not an executable file, so I corrected it.

**Comment by blag** [comment]
EL7 end-to-end tests are now failing due to Python 2/3 issues:

```
not ok 36 packs.setup_virtualenv with python3 flag works
# (in test file cli/test_pack_python3.bats, line 88)
#   `RESULT=$(st2 run examples.python_runner_print_python_version -j)' failed
not ok 37 python3 imports work correctly
# (from function `assert_success' in file cli/../test_helpers/bats-assert/src/assert.bash, line 114,
#  in test file cli/test_pack_python3.bats, line 117)
#   `assert_success' failed
# 
# -- command failed --
# status : 1
# output (19 lines):
#   
#   {
#       \"action\": {
#           \"ref\": \"python3_test.test_stdlib_import\"
#       }, 
#       \"context\": {
#           \"user\": \"st2admin\"
#       }, 
#       \"end_timestamp\": \"2020-11-18T04:43:06.745006Z\", 
#       \"id\": \"5fb4a65a4c2a7dcce5963bfe\", 
#       \"result\": {
#           \"exit_code\": 1, 
#           \"result\": \"None\", 
#           \"stderr\": \"Traceback (most recent call last):\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/netaddr/compat.py\\\", line 91, in <module>\
    from importlib import resources as _importlib_resources\
ImportError: cannot import name 'resources'\
\
During handling of the above exception, another exception occurred:\
\
Traceback (most recent call last):\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/python_runner/python_action_wrapper.py\\\", line 53, in <module>\
    from st2common import log as logging\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/log.py\\\", line 32, in <module>\
    from st2common.logging.handlers import FormatNamedFileHandler\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/logging/handlers.py\\\", line 22, in <module>\
    from oslo_config import cfg\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/oslo_config/cfg.py\\\", line 336, in <module>\
    from oslo_config import types\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/oslo_config/types.py\\\", line 21, in <module>\
    import netaddr\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/netaddr/__init__.py\\\", line 18, in <module>\
    from netaddr.core import (AddrConversionError, AddrFormatError,\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/netaddr/core.py\\\", line 11, in <module>\
    from netaddr.compat import _callable, _iter_dict_keys\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/netaddr/compat.py\\\", line 93, in <module>\
    import importlib_resources as _importlib_resources\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/importlib_resources/__init__.py\\\", line 5, in <module>\
    from ._common import (\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/importlib_resources/_common.py\\\", line 9, in <module>\
    from ._compat import (\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/importlib_resources/_compat.py\\\", line 52, in <module>\
    from typing import runtime_checkable  # type: ignore\
  File \\\"/opt/stackstorm/st2/lib/python2.7/site-packages/typing.py\\\", line 782, in <module>\
    AnyStr = TypeVar('AnyStr', bytes, unicode)\
NameError: name 'unicode' is not defined\
\", 
#           \"stdout\": \"\"
#       }, 
#       \"start_timestamp\": \"2020-11-18T04:43:06.273452Z\", 
#       \"status\": \"failed\"
#   }
# --
# 
```

**Comment by armab** [comment]
I think the e2e tests should be fixed at this point.
@nzlosh please update the Changelog to fix the git conflict. That'll re-trigger the CI as well.

**Pull request merged**
